lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
Ah, I see where the confusion comes from. I rather meant exactly put the throwing part into `allocate` (instead of silently creating an empty buffer). What are the potential pitfalls of that?